tests/test.c

```diff
@@ -527,6 +527,68 @@ void check_run_mode(robot_desc_val_t expected_run_mode) {
     print_pass();
 }
 
+// ***************************** START POS CHECK ***************************** //
+
+void check_start_pos(robot_desc_val_t expected_start_pos) {
+    // Read current start pos
+    robot_desc_val_t curr_start_pos = robot_desc_read(START_POS);
+    if (curr_start_pos != expected_start_pos) {
+        print_fail();
+        fprintf_delimiter(stderr, "Expected Start Pos:");
+        fprintf(stderr, "%s\n", (expected_start_pos == LEFT) ? "LEFT" : "RIGHT");
+        fprintf_delimiter(stderr, "Got:");
+        fprintf(stderr, "%s\n", (curr_start_pos == LEFT) ? "LEFT" : "RIGHT");
+        end_test();
+        exit(1);
+    }
+    print_pass();
+}
+
+// *************************** SUBSCRIPTION CHECK **************************** //
+
+void check_sub_requests(uint64_t dev_uid, uint32_t expected_sub_map, process_t process) {
+    // Read current subscriptions by the specified process
+    uint32_t curr_sub_map[MAX_DEVICES + 1];
+    get_sub_requests(curr_sub_map, process);
+    // Get the shm idx of the specified device
+    dev_id_t dev_ids[MAX_DEVICES];
+    uint32_t catalog;
+
+    get_device_identifiers(dev_ids);
+    get_catalog(&catalog);
+    int shm_idx = -1;
+    if (catalog != 0) {
+        // Iterate through devices
+        for (int i = 0; i < MAX_DEVICES; i++) {
```

Comment:
You should instead use the `get_dev_ix_from_uid` from `shm_wrapper`. Currently its `static`, but you can make it public and add it to the header file.

Reply:
I'll add it since I'm already working on the branch.